bug: ThumbnailWorker が例外発生時に error record を作成しない

## 概要

ThumbnailWorker が例外発生時にエラーレコードをDBに作成しないため、テスト `test_thumbnail_exception_creates_error_record` が失敗する。

## 失敗テスト

- `tests/integration/gui/workers/test_worker_error_recording.py::test_thumbnail_exception_creates_error_record`

## エラー内容

\`\`\`
AssertionError: error_count > 0 - expected > 0 but got 0
\`\`\`

## 根本原因

`src/lorairo/gui/workers/thumbnail_worker.py:217-222`:

\`\`\`python
except Exception as e:
    logger.error(f"サムネイル生成失敗: {path}", exc_info=True)
    # ワーカースレッドではDB保存不可（スレッドセーフティ）、ログのみ
    return None
\`\`\`

コメントは「ワーカースレッドではDB保存不可」だが、AnnotationWorker 等は `db_manager.repository.create_annotation_error()` を呼んでいる (Issue #108 の修正方針と同様)。ThumbnailWorker も同様に error record を保存すべき。

## 修正方針

AnnotationWorker の現行パターン (Issue #108, #107 で整理済み) に合わせ、ThumbnailWorker でも例外発生時にエラーレコードを作成する:

\`\`\`python
except Exception as e:
    logger.error(f"サムネイル生成失敗: {path}", exc_info=True)
    try:
        self.db_manager.repository.create_processing_error(
            image_id=image_id,
            operation="thumbnail_generation",
            error_message=str(e),
        )
    except Exception as db_err:
        logger.error(f"エラーレコード保存失敗: {db_err}", exc_info=True)
    return None
\`\`\`

スレッドセーフティは SQLAlchemy Session を worker 単位で持つ (またはシグナル経由でメインスレッドに委譲する) 既存設計で解決済み。

## 関連ファイル

- `src/lorairo/gui/workers/thumbnail_worker.py:217-222`
- `tests/integration/gui/workers/test_worker_error_recording.py:240-281`

## 関連 Issue

- #108 AnnotationWorker レイヤー修正 (同種パターン)
- #107 error table content 修正

--- a/src/lorairo/gui/workers/thumbnail_worker.py
+++ b/src/lorairo/gui/workers/thumbnail_worker.py
@@ -216,9 +216,15 @@ class ThumbnailWorker(LoRAIroWorkerBase[ThumbnailLoadResult]):
 
             except Exception as e:
                 batch_failed += 1
-                # ワーカースレッドではDB保存不可（スレッドセーフティ）、ログのみ
                 logger.error(
                     f"サムネイル読み込みエラー: image_id={image_id}, path={thumbnail_path}, error={e}"
+                )
+                self.db_manager.save_error_record(
+                    operation_type="thumbnail",
+                    error_type=type(e).__name__,
+                    error_message=str(e),
+                    image_id=image_id,
+                    file_path=str(thumbnail_path) if thumbnail_path else None,
                 )
 
         return batch_loaded, batch_failed

